onnx/optimizer/optimize.cc

```diff
@@ -1,45 +0,0 @@
-// ATTENTION: The code in this file is highly EXPERIMENTAL.
```

Comment:
In addition to files under `onnx/optimizer`, there are other files under `onnx/common` which we can possibly remove. For example, can we remove `onnx/common/ir.h`. I am not completely sure, but I think optimizer was the only component that was using the data structures in this files. Others that we should check are:
`#include "onnx/common/array_ref.h"
#include "onnx/common/assertions.h"
#include "onnx/common/interned_strings.h"
#include "onnx/common/graph_node_list.h"
#include "onnx/common/tensor.h"
`****

Reply:
Thank you for pointing out. However, [onnx/version_converter/helper.h](https://github.com/onnx/onnx/blob/master/onnx/version_converter/helper.h) uses `onnx/common/ir.h` as well so we might still keep them.

Reply:
Yes, some people are still using the version_converter so it makes sense to keep that functionality for now.

Reply:
I think ir.h is also out-of-date, and it is the reason of some optimizer problems: https://github.com/onnx/onnx/pull/2247. Is there any plan to update it?